crates/bevy_reflect/src/impls/glam.rs

```diff
@@ -1,44 +1,195 @@
 use crate as bevy_reflect;
+use crate::reflect::Reflect;
 use crate::ReflectDeserialize;
-use bevy_reflect_derive::{impl_from_reflect_value, impl_reflect_value};
+use bevy_reflect_derive::{
+    impl_from_reflect_value, impl_reflect_struct_and_from_reflect_struct, impl_reflect_value,
+};
 use glam::*;
 
-impl_reflect_value!(IVec2(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(IVec3(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(IVec4(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(UVec2(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(UVec3(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(UVec4(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(Vec2(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(Vec3(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(Vec3A(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(Vec4(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(DVec2(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(DVec3(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(DVec4(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(Mat3(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(Mat4(PartialEq, Serialize, Deserialize));
+impl_reflect_struct_and_from_reflect_struct!(
+    Constructor(Default::default())
```

Comment:
I'm wondering if this would be better as a separate container attribute in a manner similar to #4140 where we can define something like `#[reflect(ctor = "Default::default()")]`.